opentracing-flowid/opentracing-flowid/src/main/java/org/zalando/opentracing/flowid/DefaultFlow.java

```diff
@@ -59,7 +59,10 @@ public String currentId() {
 
     @Override
     public void writeTo(final BiConsumer<String, String> writer) {
-        writer.accept(Header.FLOW_ID, currentId());
+        extractor.extract(activeSpan())
```

Comment:
would it make sense to change the other `writeTo` method in the same way?

Reply:
You mean the `write(BiFunction<String, String, T>)` method? That one returns a `T` which is coming from the passed BiFunction.
I guess I could return null when we decide to not call it, but that might be as surprising to callers as directly throwing the IllegalStateException.